File: src/Core/Houston.Application/ViewModel/PipelineInstructionViewModels/PipelineInstructionViewModel.cs
namespace Houston.Application.ViewModel.PipelineInstructionViewModels {
	public class PipelineInstructionViewModel {
		public Guid Id { get; set; }

		public Guid PipelineId { get; set; }

		public Guid ConnectorFunctionHistoryId { get; set; }

		public int? ConnectedToArrayIndex { get; set; }

		public List<PipelineInstructionInputViewModel> PipelineInstructionInputs { get; set; } = new();
	}
}

File: src/Web/WebAngular/src/app/domain/view-models/pipeline-instruction.view-model.ts
import { PipelineInstructionInputViewModel } from "./pipeline-instruction-input.view-model";

export interface PipelineInstructionViewModel {
	id: string;
	pipelineId: string;
	connectorFunctionId: string;
	connectorFunctionHistoryId: string;
	connectedToArrayIndex: number | null;
	pipelineInstructionInputs: PipelineInstructionInputViewModel[];
}
